packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -133,13 +493,100 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+            handleOnChange(e, newValue);
+            /* @conditional-compile-remove(mention) */
+            return;
+            onChange(e, newValue);
+          }}
+          /* @conditional-compile-remove(mention) */
+          onSelect={(e) => {
+            if (caretIndex !== null) {
+              e.currentTarget.setSelectionRange(caretIndex, caretIndex);
+              setCaretIndex(null);
+              return;
+            }
+            //TODO: need to check to navigate before/after space correctly in tag + when selecting by mouse
+            /* @conditional-compile-remove(mention) */
+            if (
+              shouldHandleOnMouseDownDuringSelect &&
+              e.currentTarget.selectionStart !== null &&
+              e.currentTarget.selectionStart === e.currentTarget.selectionEnd
+            ) {
+              // handle mention click
+              const mentionTag = findMentionTagForSelection(tagsValue, e.currentTarget.selectionStart);
+              if (mentionTag !== undefined && mentionTag.plainTextBeginIndex !== undefined) {
+                if (e.currentTarget.selectionDirection === null) {
+                  e.currentTarget.setSelectionRange(
+                    mentionTag.plainTextBeginIndex,
+                    mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex
+                  );
+                } else {
+                  e.currentTarget.setSelectionRange(
+                    mentionTag.plainTextBeginIndex,
+                    mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex,
+                    e.currentTarget.selectionDirection
+                  );
+                }
+                setSelectionStartValue(mentionTag.plainTextBeginIndex);
+                setSelectionEndValue(mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex);
+              } else {
+                setSelectionStartValue(e.currentTarget.selectionStart);
+                setSelectionEndValue(e.currentTarget.selectionEnd);
+              }
+            } else {
+              updateSelectionIndexesWithMentionIfNeeded(e);
+            }
+            /* @conditional-compile-remove(mention) */
+            setShouldHandleOnMouseDownDuringSelect(false);
+          }}
+          // onMouseMove={(e) => {
+          //   console.log('!!!!not equal onMouseMove');
+          // should preventDefault be used?
+          // for handling mouse actions
+          /* @conditional-compile-remove(mention) */
+          // console.log('updateHTML onMouseMove selectionStart', e.currentTarget.selectionStart);
+          // console.log('updateHTML onMouseMove selectionEnd', e.currentTarget.selectionEnd);
+          // console.log('updateHTML onMouseMove selectionDirection', e.currentTarget.selectionDirection);
+          // if (e.currentTarget.selectionStart !== e.currentTarget.selectionEnd) {
+          //   console.log('updateHTML onMouseMove e.currentTarget.selectionStart !== e.currentTarget.selectionEnd');
```

Comment:
Remove comments

Reply:
All of them will be removed in the next PRs for this feature as this is WIP right now